`mergeOthers` example from docs doesn't work

## Bug Report

The example below from the docs doesn't work.

https://github.com/RebeccaStevens/deepmerge-ts/blob/4a73737203cb0a0e4703ea50c4f91b44c740eca1/docs/deepmergeCustom.md?plain=1#L185-L235

### Expected behavior

No type errors.

### Actual behavior

```
example.ts:20:44 - error TS2554: Expected 2 arguments, but got 1.

20         return utils.defaultMergeFunctions.mergeOthers(values);
                                              ~~~~~~~~~~~

  node_modules/.pnpm/deepmerge-ts@7.0.2/node_modules/deepmerge-ts/dist/node/index.d.mts:626:95
    626 declare function mergeOthers<Ts extends ReadonlyArray<unknown>>(m_target: Reference<unknown>, values: Ts): void;
                                                                                                      ~~~~~~~~~~
    An argument for 'values' was not provided.


Found 1 error in example.ts:20
```

Minimal reproduction: https://github.com/nvlang/deepmerge-ts-bug

---

I noticed this after getting some linter warnings/errors after updating `deepmerge-ts` from `5.1.0` to `7.0.2`, because I used `deepmergeCustom` to create a merge function which filters out undefined and null values, and doesn't merge arrays (overriding them instead):

```ts
export const mergeCustom = deepmergeCustom({
    mergeOthers: (values, utils) =>
        utils.defaultMergeFunctions.mergeOthers(
            values.filter((v) => v !== undefined && v !== null),
        ),
    // Don't merge arrays
    mergeArrays: (values) => values[values.length - 1],
});
```

I checked the CHANGELOG.md for breaking changes, and saw a note about the generics around deepmergeCustom having changed:

> The order of the generics of `deepmergeCustom` and `deepmergeIntoCustom` have changed. If you are passing generics to these functions you need to update them. [[Source](https://github.com/RebeccaStevens/deepmerge-ts/blob/4a73737203cb0a0e4703ea50c4f91b44c740eca1/CHANGELOG.md#breaking-changes-1)]

However, in my case I wasn't passing any type parameters, so I didn't think I'd have to change anything.

Clarification regarding this would be much appreciated, but, in any event, thank you for your work on a great library!

--- a/src/deepmerge-into.ts
+++ b/src/deepmerge-into.ts
@@ -3,7 +3,7 @@ import {
   defaultFilterValues,
   defaultMetaDataUpdater,
 } from "./defaults/general";
-import * as defaultMergeIntoFunctions from "./defaults/into";
+import { mergeIntoFunctions as defaultMergeIntoFunctions } from "./defaults/into";
 import {
   type DeepMergeBuiltInMetaData,
   type DeepMergeFunctionsDefaultURIs,

--- a/src/deepmerge.ts
+++ b/src/deepmerge.ts
@@ -3,7 +3,7 @@ import {
   defaultFilterValues,
   defaultMetaDataUpdater,
 } from "./defaults/general";
-import * as defaultMergeFunctions from "./defaults/vanilla";
+import { mergeFunctions as defaultMergeFunctions } from "./defaults/vanilla";
 import {
   type DeepMergeBuiltInMetaData,
   type DeepMergeFunctionsDefaultURIs,

--- a/src/defaults/into.ts
+++ b/src/defaults/into.ts
@@ -10,11 +10,11 @@ import { getIterableOfIterables, getKeys, objectHasProperty } from "../utils";
  * The default merge functions.
  */
 export type MergeFunctions = {
-  mergeRecords: typeof mergeRecords;
-  mergeArrays: typeof mergeArrays;
-  mergeSets: typeof mergeSets;
-  mergeMaps: typeof mergeMaps;
-  mergeOthers: typeof mergeOthers;
+  mergeRecords: typeof mergeRecordsInto;
+  mergeArrays: typeof mergeArraysInto;
+  mergeSets: typeof mergeSetsInto;
+  mergeMaps: typeof mergeMapsInto;
+  mergeOthers: typeof mergeOthersInto;
 };
 
 /**
@@ -23,7 +23,7 @@ export type MergeFunctions = {
  * @param m_target - The result will be mutated into this record
  * @param values - The records (including the target's value if there is one).
  */
-export function mergeRecords<
+function mergeRecordsInto<
   Ts extends ReadonlyArray<Record<PropertyKey, unknown>>,
   U extends DeepMergeIntoFunctionUtils<M, MM>,
   M,
@@ -79,7 +79,7 @@ export function mergeRecords<
  * @param m_target - The result will be mutated into this array
  * @param values - The arrays (including the target's value if there is one).
  */
-export function mergeArrays<Ts extends ReadonlyArray<ReadonlyArray<unknown>>>(
+function mergeArraysInto<Ts extends ReadonlyArray<ReadonlyArray<unknown>>>(
   m_target: Reference<unknown[]>,
   values: Ts,
 ): void {
@@ -92,7 +92,7 @@ export function mergeArrays<Ts extends ReadonlyArray<ReadonlyArray<unknown>>>(
  * @param m_target - The result will be mutated into this set
  * @param values - The sets (including the target's value if there is one).
  */
-export function mergeSets<
+function mergeSetsInto<
   Ts extends ReadonlyArray<Readonly<ReadonlySet<unknown>>>,
 >(m_target: Reference<Set<unknown>>, values: Ts): void {
   for (const value of getIterableOfIterables(values.slice(1))) {
@@ -106,7 +106,7 @@ export function mergeSets<
  * @param m_target - The result will be mutated into this map
  * @param values - The maps (including the target's value if there is one).
  */
-export function mergeMaps<
+function mergeMapsInto<
   Ts extends ReadonlyArray<Readonly<ReadonlyMap<unknown, unknown>>>,
 >(m_target: Reference<Map<unknown, unknown>>, values: Ts): void {
   for (const [key, value] of getIterableOfIterables(values.slice(1))) {
@@ -117,9 +117,20 @@ export function mergeMaps<
 /**
  * Set the target to the last non-undefined value.
  */
-export function mergeOthers<Ts extends ReadonlyArray<unknown>>(
+function mergeOthersInto<Ts extends ReadonlyArray<unknown>>(
   m_target: Reference<unknown>,
   values: Ts,
 ) {
   m_target.value = values.at(-1);
 }
+
+/**
+ * The merge functions.
+ */
+export const mergeIntoFunctions = {
+  mergeRecords: mergeRecordsInto,
+  mergeArrays: mergeArraysInto,
+  mergeSets: mergeSetsInto,
+  mergeMaps: mergeMapsInto,
+  mergeOthers: mergeOthersInto,
+};

--- a/src/defaults/vanilla.ts
+++ b/src/defaults/vanilla.ts
@@ -27,7 +27,7 @@ export type MergeFunctions = {
  *
  * @param values - The records.
  */
-export function mergeRecords<
+function mergeRecords<
   Ts extends ReadonlyArray<Record<PropertyKey, unknown>>,
   U extends DeepMergeUtils<M, MM>,
   Fs extends DeepMergeFunctionsURIs,
@@ -88,7 +88,7 @@ export function mergeRecords<
  *
  * @param values - The arrays.
  */
-export function mergeArrays<
+function mergeArrays<
   Ts extends ReadonlyArray<ReadonlyArray<unknown>>,
   Fs extends DeepMergeFunctionsURIs,
   M,
@@ -101,9 +101,9 @@ export function mergeArrays<
  *
  * @param values - The sets.
  */
-export function mergeSets<
-  Ts extends ReadonlyArray<Readonly<ReadonlySet<unknown>>>,
->(values: Ts): DeepMergeSetsDefaultHKT<Ts> {
+function mergeSets<Ts extends ReadonlyArray<Readonly<ReadonlySet<unknown>>>>(
+  values: Ts,
+): DeepMergeSetsDefaultHKT<Ts> {
   return new Set(getIterableOfIterables(values)) as DeepMergeSetsDefaultHKT<Ts>;
 }
 
@@ -112,7 +112,7 @@ export function mergeSets<
  *
  * @param values - The maps.
  */
-export function mergeMaps<
+function mergeMaps<
   Ts extends ReadonlyArray<Readonly<ReadonlyMap<unknown, unknown>>>,
 >(values: Ts): DeepMergeMapsDefaultHKT<Ts> {
   return new Map(getIterableOfIterables(values)) as DeepMergeMapsDefaultHKT<Ts>;
@@ -121,6 +121,17 @@ export function mergeMaps<
 /**
  * Get the last non-undefined value in the given array.
  */
-export function mergeOthers<Ts extends ReadonlyArray<unknown>>(values: Ts) {
+function mergeOthers<Ts extends ReadonlyArray<unknown>>(values: Ts) {
   return values.at(-1);
 }
+
+/**
+ * The merge functions.
+ */
+export const mergeFunctions = {
+  mergeRecords,
+  mergeArrays,
+  mergeSets,
+  mergeMaps,
+  mergeOthers,
+};

